### ISSUE TITLE
Pass variant value to styled component

### ISSUE THREAD

**Issue by JamesHemery** [nandorojo/dripsy#184]
Hi :) 

Actually the variant/variants props is not passed to a styled component.

**Example:**

```js
function Button({title, variant}){
   // Here variant is undefined
}

export default styled(Button, {
  defaultVariant: 'default',
  themeKey: 'button',
})();
```

If I render `<Button variant="primary" title="Login"/>` the variant Button component can't access to the variant props, in some case this is uncomfortable.

In my case, I need the variant to condition some props in a Moti animation.

Do you agree to merge a PR that pass the variant/variants prop ?

Thank you in advance,

**Comment by nandorojo** [comment]
Interesting, I'm not sure if that should happen. It seems like you're trying to circumvent Dripsy handling the variant itself, but it's an internal prop, not exactly one that gets forwarded down. Are you using this to style text components?

If I can better understand your use-case, it would be easier to understand the benefit on my end. Thanks!

**Comment by JamesHemery** [comment]
@nandorojo In my case, it is a Button component on which I have a Moti animation. I vary the color of the text, and currently I can only do it in the component.

I think that having access to the variant / variants within the component is important in any case since it is in a way the identity of the component that changes too.

Thanks

**Comment by nandorojo** [comment]
got it, makes sense. in my case, i implement the variant styles directly in the component, and pass it down with context: https://github.com/nandorojo/dripsy/discussions/162#discussioncomment-1901320

I think this makes sense though. I may have to make some edits to the PR.

What may make more sense is if your `Button` component renders a `ButtonContainer` which receives the `variant` passed to `Button`, though.

**Comment by gorbypark** [comment]
I think I also have a use case for this (more for ergonomics than anything).  Say I have a button component with two variants, primary and disabled.  If I set the disabled variant, `<Button variant="disabled" />`, inside the Button component I'd like to set a `disabled={variant === 'disabled'}` prop on a Pressable to disable the onPress.  Right now I'd have to do `<Button variant="disabled" disabled={true} />` and pass a disabled prop into the Button component.  It would be nice not to have to "set it twice" and only require passing in the variant and handle the disabling inside the component itself based on the variant.

**Comment by nandorojo** [comment]
Apologies for the delay on this. I have released it in 4.2.0. Components now receive the `variant` prop, and it falls back to `defaultVariant`.
